source/loader/layers/sanitizer/asan_report.cpp

```diff
@@ -131,27 +132,35 @@ void ReportUseAfterFree(const DeviceSanitizerReport &Report,
     context.logger.always("  #0 {} {}:{}", Func, File, Report.Line);
     context.logger.always("");
 
-    auto AllocInfoItOp =
-        context.interceptor->findAllocInfoByAddress(Report.Address);
-    if (!AllocInfoItOp) {
-        context.logger.always("Failed to find which chunck {} is allocated",
-                              (void *)Report.Address);
-    } else {
-        auto &AllocInfo = (*AllocInfoItOp)->second;
-        if (AllocInfo->Context != Context) {
+    if (Options().MaxQuarantineSizeMB > 0) {
+        auto AllocInfoItOp =
+            context.interceptor->findAllocInfoByAddress(Report.Address);
+
+        if (!AllocInfoItOp) {
             context.logger.always("Failed to find which chunck {} is allocated",
                                   (void *)Report.Address);
+        } else {
+            auto &AllocInfo = (*AllocInfoItOp)->second;
```

Comment:
Do you need to assert AllocInfo is not NULL ?

Reply:
AllocInfo will not be NULL here. If not found a right AllocInfo for address, AllocInfoItOp will set to no_value, and will not go to this branch.